### PR TITLE
ci: add CI to run through all build tools

### PR DESCRIPTION
**Pull request by grant**
## ci: add CI to run through all build tools

- Runs jsonschema/registry tools on CI. Ensures running the tools produce a 0 diff.
- Splits schema validator workflow with build workflow.

---

- Requires: https://github.com/googleapis/google-cloudevents/pull/115 (Note: diff will be a bit smaller once that PR is merged.)
- Fixes: https://github.com/googleapis/google-cloudevents/issues/116

### CODE REVIEW

**Comment by grant** [comment]
Hey @jskeet, @grayside, this PR should be ready.

It will ensure that the tools are run whenever we make a change to a proto or tool itself. Also makes the order of running these tools clear.

File: jsonschema/catalog.json
Comment (jskeet) [inline jsonschema/catalog.json:15]: Why are all these changes here? Is the ordering non-deterministic at the moment, or was it non-deterministic before and this just corrects it?

File: .github/workflows/generate.yaml
Comment (jskeet) [inline .github/workflows/generate.yaml:1]: Will this (and Validate) be run even if Compile fails? Can we make it wait until Compile succeeds before this is run?

(This is why I prefer them being separate steps in one action - while the PR description says it "Also makes the order of running these tools clear" it feels to me like it's doing the opposite.)

File: jsonschema/catalog.json
Comment (grant) [reply jsonschema/catalog.json:15]: It's currently non-deterministic. I state that the PR will be smaller due to this issue in the PR description.

I need to merge https://github.com/googleapis/google-cloudevents/pull/115 before this PR to not see this artifact.

File: .github/workflows/generate.yaml
Comment (grant) [reply .github/workflows/generate.yaml:1]: I can merge them in one large action, although I would rather see 1/2 checks pass or 2/2 checks pass than 0/1 checks pass in the UI. Yes, the 2nd workflow (generate) cannot pass if the 1st workflow fails (compile).

The order is clearer as we have 4 tools in https://github.com/googleapis/google-cloudevents/tree/master/tools and we do not describe the specific order required to run them to correctly generate results. This CI does that, as you can just look at the CI and run the commands rather than looking at each tool.

File: jsonschema/catalog.json
Comment (jskeet) [reply jsonschema/catalog.json:15]: Ah, right. I'd missed that.

File: .github/workflows/generate.yaml
Comment (jskeet) [reply .github/workflows/generate.yaml:1]: I think it's subjective, and I guess you'll be working in this repo more than I will be. I think the order is clearest if there's just *one* action that takes all the steps, but it's your call.

**Review by jskeet — APPROVED** [review]
Approved contingent on #115 being merged, etc.

**Pull request merged**
